src/planwise/component/scenarios.clj

```diff
@@ -27,20 +28,43 @@
   [changeset]
   (apply +' (map :investment changeset)))
 
+(defn- create-text-description
+  [changeset]
+  (let [sites         (count changeset)
+        capacity      (apply +' (mapv #(-> % (read-string) :capacity) changeset))
+        text-default  ["Create replace site." "Increase overall capacity in replace."]]
+    (str/join " "
+              (map (fn [data message] (cond (pos? data) (str/replace message #"replace" (str data))))
+                   [sites capacity] (if (= sites 1) text-default (assoc text-default 0 "Create replace sites."))))))
+
 ;; ----------------------------------------------------------------------
 ;; Service definition
 
-(defn list-scenarios
-  [store project-id]
-  ;; TODO compute % coverage from initial scenario/project
-  (db-list-scenarios (get-db store) {:project-id project-id}))
-
 (defn get-scenario
   [store scenario-id]
   ;; TODO compute % coverage from initial scenario/projects
   (-> (db-find-scenario (get-db store) {:id scenario-id})
       (update :changeset edn/read-string)))
 
+(defn- get-scenario-state
+  [store scenario-id]
+  (:state (get-scenario store scenario-id)))
+
+(defn get-changeset-resume
+  [store scenario-id]
+  (let [{:keys [changeset]} (get-scenario store scenario-id)]
+    (create-text-description (map pr-str changeset))))
+
+(defn list-scenarios
+  [store project-id]
+  ;; TODO compute % coverage from initial scenario/project
+  (let [list (db-list-scenarios (get-db store) {:project-id project-id})]
+    (map (fn [scenario] (-> scenario
```

Comment:
add the `state` field to `db-list-scenarios` so there is no need for `get-scenario-stat`.
remove `get-changeset-resume` and use directly `build-changeset-summary`